### DISCUSSION TITLE
Clipboard question in ios.

### DISCUSSION THREAD

**Discussion by schyou1** [FreeRDP/FreeRDP#7478] [Q&A]
developing clipboard in ios.

Add the clipboard and the clipboard channel is connected.
After that, the "Server Clipboard Function PDU" does not come.

### ㅇ add clipboard
    addArgument(&argc, &argv, "/clipboard");

### ㅇ clipboard channel is connected.
     static void ios_OnChannelConnectedEventHandler(void *context, ChannelConnectedEventArgs *e) {
         if (strcmp(e->name, CLIPRDR_SVC_CHANNEL_NAME) == 0) {
          ios_cliprdr_init(afc, (CliprdrClientContext*)e->pInterface);
        }
     }

      BOOL ios_cliprdr_init(mfContext* mfc, CliprdrClientContext* cliprdr) {
          wClipboard* clipboard;

          if (!mfc || !cliprdr)
             return FALSE;

           if (!(clipboard = ClipboardCreate())) {
             return FALSE;
          }
           mfc->cliprdr = cliprdr;
           mfc->clipboard = clipboard;
           cliprdr->custom = (void*)mfc;
           cliprdr->MonitorReady = wf_cliprdr_monitor_ready;
           cliprdr->ServerCapabilities = wf_cliprdr_server_capabilities;
           cliprdr->ServerFormatList = wf_cliprdr_server_format_list;
           cliprdr->ServerFormatListResponse = wf_cliprdr_server_format_list_response;
           cliprdr->ServerLockClipboardData = wf_cliprdr_server_lock_clipboard_data;
           cliprdr->ServerUnlockClipboardData = wf_cliprdr_server_unlock_clipboard_data;
           cliprdr->ServerFormatDataRequest = wf_cliprdr_server_format_data_request;
           cliprdr->ServerFormatDataResponse = wf_cliprdr_server_format_data_response;
           cliprdr->ServerFileContentsRequest = wf_cliprdr_server_file_contents_request;
           cliprdr->ServerFileContentsResponse = wf_cliprdr_server_file_contents_response;

           return TRUE;
     }

### ㅇ clipboard

                                     Initialization Sequence\n
      Client                                                                    Server\n
         |                                                                         |\n
         |<----------------------Server Clipboard Capabilities PDU-----------------|\n
         |<-----------------------------Monitor Ready PDU--------------------------|\n
         |-----------------------Client Clipboard Capabilities PDU---------------->|\n
         |---------------------------Temporary Directory PDU---------------------->|\n
         |-------------------------------Format List PDU-------------------------->|\n
         |<--------------------------Format List Response PDU----------------------|\n
 
Please help

**Comment by akallabeth** [comment]
1. Some more context (or source to look at?) would be very helpful.
1. Only thing that comes to mind are:
   1.  does your event loop work?
   1. does the server have `clipboard` enabled?
